airgun/entities/host_new.py

```diff
@@ -32,6 +38,163 @@ def get_details(self, entity_name, widget_names=None):
+            release_ver: Release version to be assigned
+            add_ons: Add-ons to be assigned
+
+        Raises:
+            ValueError: If no parameters are passed.
+        """
+
+        if not any([role, sla, usage, release_ver, add_ons]):
+            raise ValueError(
+                'At least one of the role, sla, usage, release_ver, add_ons ' 'must be provided!'
+            )
+        view = self.navigate_to(self, 'NewDetails', entity_name=entity_name)
+        view.wait_displayed()
+        view.overview.system_purpose.edit_system_purpose.click()
+        view = EditSystemPurposeView(self.browser)
+        self.browser.plugin.ensure_page_safe()
+        view.wait_displayed()
+        if role:
+            view.role.fill(role)
+        if sla:
+            view.sla.fill(sla)
+        if usage:
+            view.usage.fill(usage)
+        if release_ver:
+            view.release_version.fill(release_ver)
+        if add_ons:
+            view.add_ons.fill(add_ons)
+        view.save.click()
+
+    def add_host_to_host_collection(
+        self, entity_name, host_collection_name=None, add_to_all_collections=False
+    ):
+        """
+        Function that adds host to host collection
+
+        Args:
+            entity_name: Name of the host
+            host_collection_name: Name of the host collection we want to remove host from
+            add_to_all_collections: If True, host will be added to all host collections
+
+        Raises:
+            ValueError: If specific hostColName is set to be removed and add all is set to True.
+            ValueError: If no parameters are passed.
+            ValueError: If host is already assigned to selected host collection.
+            ValueError: If there are no host collections left for addition.
+            ValueError: Given host collection name is not found in host collections.
+        """
+
+        if host_collection_name and add_to_all_collections:
+            raise ValueError(
+                'Either host_collection_name or add_to_all_collections must be provided!'
+            )
+        if (host_collection_name is None) and (add_to_all_collections is False):
+            raise ValueError(
+                'host_collection_name None and add_to_all_collections False '
+                'is invalid combination!'
+            )
+
+        view = self.navigate_to(self, 'NewDetails', entity_name=entity_name)
+        view.wait_displayed()
+        self.browser.plugin.ensure_page_safe()
+        # Handle the case where there are no host collections assigned
+        if view.overview.host_collections.no_host_collections.is_displayed:
+            view.overview.host_collections.add_to_host_collection.click()
+        else:
+            if (
+                host_collection_name
+                in view.overview.host_collections.assigned_host_collections.read()
+            ):
+                raise ValueError(f'{host_collection_name} already assigned to host!')
```

Comment:
Food for thought, is that really an error?

Reply:
Good question. Probably not, but I think that the user of that function has the intention of adding the host to the new host collection and then checking if the host was actually added to that collection. So he should be notified that it didn't go according to his intentions in my opinion.

Reply:
Yes, you're right, especially since this is used for testing - we should be careful and explicit about those things.